pandas/core/frame.py

```diff
@@ -368,14 +368,18 @@ def _constructor(self):
     _accessors = set()  # type: Set[str]
 
     @property
-    def _constructor_expanddim(self):
+    def _constructor_expanddim(self) -> None:
```

Comment:
This technically should be a NoReturn but we can't use that for Py35 compat just yet. Probably better to remove this for now

Reply:
Yeah `NoReturn` is the proper type here. But I think it is available in python 3.5 https://docs.python.org/3/library/typing.html#typing.NoReturn or am I missing something.

Reply:
Right but not until 3.5.4 - we'd still have to bump to that if we wanted this. Not sure this annotation alone is worth doing that

Reply:
Oh, ok. I thought we are supporting 3.5.4